bittide-instances/src/Bittide/Instances/Hitl/LinkConfiguration.hs

```diff
@@ -0,0 +1,222 @@
+  (txns, txps, v2bv . fmap boolToBit <$> linkUps, stats, spiDone, spiOut)
+ where
+  sysRst = rst `orReset` unsafeFromActiveLow (fmap not spiErr)
+
+  -- Clock programming
+  spiDone = E.dflipflop sysClk $ (== Finished) <$> spiState
+  spiErr = E.dflipflop sysClk $ isErr <$> spiState
+
+  isErr = \case
+    Error {} -> True
+    _        -> False
+
+  (_, _, spiState, spiOut) =
+    withClockResetEnable sysClk sysRst enableGen $
+      si539xSpi testConfig6_200_on_0a (SNat @(Microseconds 10)) (pure Nothing) miso
+
+  -- Transceiver setup
+  gthAllReset = rst `orReset` unsafeFromActiveLow spiDone
+
+  (_txClocks, rxClocks, txns, txps, linkUpsRx, stats) = unzip6 $
+    transceiverPrbsN
+      @GthTx @GthRx @Ext200 @Basic125 @GthTx @GthRx
+      refClk sysClk gthAllReset
+      channelNames clockPaths rxns rxps
+
+  syncLink rxClock linkUp = xpmCdcSingle rxClock sysClk linkUp
+  linkUps = bundle $ zipWith syncLink rxClocks linkUpsRx
+
+-- | Top entity for this test. See module documentation for more information.
+linkConfigurationTest ::
+  "SMA_MGT_REFCLK_C" ::: DiffClock Ext200 ->
+  "SYSCLK_300" ::: DiffClock Ext300 ->
+  "SYNC_IN" ::: Signal Basic125 Bool ->
+  "GTH_RX_NS" ::: TransceiverWires GthRx ->
+  "GTH_RX_PS" ::: TransceiverWires GthRx ->
+  "MISO" ::: Signal Basic125 Bit ->
+  ( "GTH_TX_NS" ::: TransceiverWires GthTx
+  , "GTH_TX_PS" ::: TransceiverWires GthTx
+  , "SYNC_OUT" ::: Signal Basic125 Bool
+  , "spiDone" ::: Signal Basic125 Bool
+  , "" :::
+      ( "SCLK"      ::: Signal Basic125 Bool
+      , "MOSI"      ::: Signal Basic125 Bit
+      , "CSB"       ::: Signal Basic125 Bool
+      )
+  )
+linkConfigurationTest refClkDiff sysClkDiff syncIn rxns rxps miso =
+  (txns, txps, syncOut, spiDone, spiOut)
+ where
+  refClk = ibufds_gte3 refClkDiff :: Clock Ext200
+  (sysClk, sysRst) = clockWizardDifferential sysClkDiff noReset
+
+  -- the test starts with a valid configuration coming in
+  startTest = isJust <$> testConfig
+
+  -- use some simple SYNC_IN / SYNC_OUT synchronization to
+  -- synchronously start the test
+  syncInRst =
+      resetGlitchFilter (SNat @1024) sysClk
+    $ unsafeFromActiveLow
+    $ xpmCdcSingle sysClk sysClk syncIn
+  syncOut = startTest
+
+  testRst =   sysRst
+    `orReset` syncInRst
+    `orReset` unsafeFromActiveLow startTest
+    `orReset` unsafeFromActiveHigh skip
+
+  -- derive the test settings from the passed configuration
+  skip = maybe False (not . fpgaEnabled) <$> testConfig
+  mask = maybe 0 disabledLinkMask <$> testConfig
+
+  (txns, txps, linkUps, _stats, spiDone, spiOut) =
+    transceiversStartAndObserve refClk sysClk testRst rxns rxps miso
+
+  -- Consider test done if enabled links have been up consistently for
+  -- 50 seconds. This is just below the test timeout of 60 seconds, so
+  -- transceivers have ~10 seconds to come online reliably. This
+  -- should be plenty.
+  endSuccess = trueFor (SNat @(Seconds 50)) sysClk testRst linkMaskMatch
```

Comment:
For this test we can assume the links will be stable after they're up. "Long" term link stability is already tested in the the transceiver up test. That way the whole suite will take ~10 seconds instead of ~10 minutes :-).